MobiusCore/Source/EffectHandlers/EffectRouter.swift

```diff
@@ -58,19 +60,23 @@ private struct Route<Input, Output> {
     let tryRoute: (Input, @escaping Consumer<Output>) -> Bool
     let disposable: Disposable
 
-    init<Payload>(
+    init<Payload, Handler: EffectHandler>(
         path tryPath: @escaping (Input) -> Payload?,
-        handler: EffectHandler<Payload, Output>
-    ) {
+        handler: Handler
+    ) where Handler.Input == Payload, Handler.Output == Output {
+        var disposables: [Disposable] = []
```

Comment:
Does not this variable scope mean that two concurrent invocations of the effect handler use the same array instance? Not necessarily concurrent on different threads; I guess depending on the semantics of handling an effect/sending events, the effect handler could be recursively invoked before it completes. Maybe that is safe. 

Also, I think the array will grow indefinitely? It seems like there needs to be a way to remove disposables once there is no longer a need for cancellation? Is it possible to write a test that verifies that there is no memory leak?

Reply:
The access to the blocks which modify this array is synchronized, so it should not be an issue.

The garbage thing is an issue though. I will write about it in a comment in the main PR thread